concent_api/core/tests/test_unit_model.py

```diff
@@ -162,3 +169,232 @@ def test_that_storing_subtask_with_task_to_compute_nested_in_another_messages_wi
+        self.deposit_account.ethereum_public_key = self.payee_ethereum_address
+        self.deposit_account.clean()
+        self.deposit_account.save()
+
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+
+        with pytest.raises(ValidationError):
+            self.deposit_claim.clean()
+            self.deposit_claim.save()
+
+    def test_that_exception_is_rised_when_payee_ethereum_public_key_is_not_equal_ethereum_address_length(self):
+        self.client.clean()
+        self.client.save()
+
+        self.deposit_account.client = self.client
+        self.deposit_account.clean()
+        self.deposit_account.save()
+
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+        self.deposit_claim.payee_ethereum_public_key = self.payee_ethereum_address + '1'
+        with pytest.raises(ValidationError):
+            self.deposit_claim.clean()
+            self.deposit_claim.save()
+
+    def test_that_exception_is_rised_when_amount_is_equal_to_zero(self):
+        self.client.clean()
+        self.client.save()
+
+        self.deposit_account.client = self.client
+        self.deposit_account.clean()
+        self.deposit_account.save()
+
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+        self.deposit_claim.amount = 0
+        with pytest.raises(ValidationError):
+            self.deposit_claim.clean()
+            self.deposit_claim.save()
+
+    def test_that_exception_is_rised_when_amount_is_less_then_zero(self):
+        self.client.clean()
+        self.client.save()
+
+        self.deposit_account.client = self.client
+        self.deposit_account.clean()
+        self.deposit_account.save()
+
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+        self.deposit_claim.amount = -1
+        with pytest.raises(ValidationError):
+            self.deposit_claim.clean()
+            self.deposit_claim.save()
+
+    def test_that_exception_is_rised_when_tx_hash_is_not_none_and_not_string(self):
+        self.client.clean()
+        self.client.save()
+
+        self.deposit_account.client = self.client
+        self.deposit_account.clean()
+        self.deposit_account.save()
+
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+        self.deposit_claim.tx_hash = 1
+        with pytest.raises(ValidationError):
+            self.deposit_claim.clean()
+            self.deposit_claim.save()
+
+    def test_that_deposit_account_is_not_removed_when_deposit_claim_is_deleted(self):
+        self.client.clean()
+        self.client.save()
+
+        self.deposit_account.client = self.client
+        self.deposit_account.clean()
+        self.deposit_account.save()
+
+        self.deposit_claim.payer_deposit_account = self.deposit_account
+
+        self.deposit_claim.clean()
+        self.deposit_claim.save()
+
+        DepositClaim.objects.first().delete()
+
+        assert DepositAccount.objects.first() is not None
```

Comment:
You can use something like `self.assertIsNot(DepositAccount.objects.first(), None)`. I don't think we use standard `assert` in tests.

Reply:
@Jakub89 Please check my comment, we should avoid checking `.first()` object as it somehow (sometimes mysteriously) might be something different, and always check by `pk` to be sure that we are checking exactly the same object.

Reply:
@rwrzesien Okey, I'll keep that in mind.